File: app/src/main/java/com/sumalogos/webster/sumalogos/apiservice/IBaseAPIService.java
package com.sumalogos.webster.sumalogos.apiservice;

import com.sumalogos.webster.sumalogos.util.LiveDataResult;

/**
 * Created by webster on 18/01/18.
 */

public interface IBaseAPIService {
    <T> LiveDataResult<?> storeResult(T object, String methodName, String tag);

    <T> LiveDataResult<T> storeError(Throwable error, String methodName, String tag);
}

File: app/src/main/java/com/sumalogos/webster/sumalogos/apiservice/IBaseAPIService.kt
package com.sumalogos.webster.sumalogos.apiservice

import com.sumalogos.webster.sumalogos.util.LiveDataResult



/**
 * Created by webster on 02/01/18.
 */

interface IBaseAPIService {
    fun <T> storeResult(`object`: T): LiveDataResult<*>

    fun <T> storeError(error: Throwable): LiveDataResult<T>
}